draft-ietf-dprive-xfr-over-tls.md

```diff
@@ -333,29 +336,186 @@ The figure below provides an outline of the AXoT mechanism including NOTIFYs.
-opened TLS connection to send a SOA request. Using an existing open connection
-is RECOMMENDED over opening a new connection.
-
-QUESTION: Should there be a requirement that the SOA is always done on a TLS
-connection if the XFR is? For the case when no transfer is required this could
-be unnecessary overhead.
+and primary to use a port other than port 853 for AXoT.
+
+### Coverage and relationship to RFC5936
+
+[@RFC5936] re-specified AXFR providing additional guidance beyond that provided
+in [@RFC1045] and [@RFC1035]. For example, sections 4.1, 4.1.1 and 4.1.2 of
+[@RFC5936] provide improved guidance for AXFR clients and servers with regard to
+re-use of connections for multiple AXFRs, AXFRs of different zones and using TCP
+connection for other queries including SOA. However [@RFC5936] was constrained
+by having to be backwards compatible with some very early basic implementations
+of AXFR. 
+
+Here we specify some optimised behaviours for AXoT, based
+closely on those in [@RFC5936], but without the constraint of backwards
+compatibility since it is expected that all implementations of AXoT fully
+implement the behaviour described here.
+
+Where any behavior is not explicitly described here, the behviour specified in
+[@RFC5953] MUST be followed. Any behaviour specified here takes precedence for
+AXoT implementations over that in [@RFC5953].
+
+### AXoT connection and message handling
+
+The first paragraph of Section 4.1.1 of [@RFC5936] says that clients SHOULD
+close the connection when there is no 'apparent need' to use the connection for
+some time period.
+
+For AXoT this requirement is updated: AXoT clients and servers SHOULD use EDNS0
+Keepalive [RFC7828] to establish the connection timeouts to be used. The AXoT
+server may use the frequency of recent AXFRs to calculate an average update rate
+as input to the decision of what EDNS0 Keepalive timeout to use. Note that use
+of EDNS0 Keepalive enables AXoT servers to signal the desire to close a
+connection due to low resources by sending an EDNS0 Keepalive option with a
+timeout of 0 (in the absence of another way to signal the abort of a AXoT
+transfer). If the server does not support EDNS0 Keepalive the client MAY keep
+the connection open for a few seconds ([@RFC7766] recommends that servers use
+timeouts of at least a few seconds).
+
+An AXoT server MUST be able to handle multiple IXFR requests on a
+single TLS connection (for the same and different zones), as well as to handle
+other query/response transactions over it.
+
+[@RFC5936] says:
+
+    "An AXFR client MAY use an already opened TCP connection to start an
+    AXFR session.  Using an existing open connection is RECOMMENDED over
+    opening a new connection.  (Non-AXFR session traffic can also use an
+    open connection.)"
+
+For AXoT this requirement is restated: AXoT clients SHOULD re-use an existing
+open TLS connection when starting any new AXoT session to the same primary, and
```

Comment:
AXoT session -> AXoT transfers?

Reply:
I used session because that is what is used in RFC5936 and it think it is helpful to clearly distinguish an AXoT session (that is one request and potentially many responses) from an IXoT transaction or transfer which is just a query/response.  RFC5936 deals with some of the special error cases around AXFR due to the 'session' nature of the exchange which I think we also need to think about a bit more.....